extensions/pyRevitTools.extension/checks/cad_audit_check.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+
+from pyrevit import script, revit, DB, DOCS
+from pyrevit.forms import alert
+from pyrevit.preflight import PreflightTestCase
+
+from System.Windows import Window # Used for cancel button
+from rpw.ui.forms import (FlexForm, Label, Separator, Button) # RPW
+from rpw.ui.forms.flexform import RpwControlMixin, Controls # Used to create RadioButton
+
+from pyrevit.coreutils import Timer # Used for timing the check
+from datetime import timedelta # Used for timing the check
+
+doc = DOCS.doc
+ac_view = doc.ActiveView
+
+
+def collect_cadinstances(active_view_only):
+    """     Collect ImportInstance class from whole model or from just active view  """
+    if active_view_only:
+        cadinstances = DB.FilteredElementCollector(doc, ac_view.Id).OfClass(DB.ImportInstance).WhereElementIsNotElementType().ToElements()
+    else:
+        cadinstances = DB.FilteredElementCollector(doc).OfClass(DB.ImportInstance).WhereElementIsNotElementType().ToElements()
+    if cadinstances:
+        return cadinstances
+    else:
+        alert("No CAD instances found in the {}.".format("active view" if active_view_only else "model"), exitscript=True)
+
+# Manage Flexform cancel using .NET System.Windows RoutedEventArgs Class
+class ButtonClass(Window): # to handle button event (https://stackoverflow.com/questions/54756424/using-the-on-click-option-for-a-revit-rpw-flexform-button)
+    @staticmethod
+    def cancel_clicked(sender, e):
+        window = Window.GetWindow(sender)
+        window.close()
+        script.exit()
+
+# Add radio button functionality to RPW Flexforms
+class RadioButton(RpwControlMixin, Controls.RadioButton):
+    """
+    Windows.Controls.RadioButton Wrapper
+
+    >>> RadioButton('Label')
+    """
+    def __init__(self, name, radio_text, default=False, **kwargs):
+        """
+        Args:
+            name (``str``): Name of control. Will be used to return value
+            radio_text (``str``): RadioButton label Text
+            default (``bool``): Sets IsChecked state [Default: False]
+            wpf_params (kwargs): Additional WPF attributes
+        """
+        self.Name = name
+        self.Content = radio_text
+        self.IsChecked = default
+        self.set_attrs(top_offset=0, **kwargs)
+
+    @property
+    def value(self):
+        return self.IsChecked
+    
+def get_cad_site(cad_inst):
+    """ A CAD's location site cannot be got from the Shared Site parameter 
+        cad_inst.Name returns the site name with a 'location' prefix (language-specific, eg 'emplacement' in French)"""
+    cad_site_name = cad_inst.Name
+    cad_site_name.replace("location", "-")
+    return cad_site_name
```

Comment:
the `str.replace()` method doesn't modify the input value, but it returns a value that needs to be saved in a variable.
I believe this should look like this:

```suggestion
    return = cad_inst.Name.replace("location", "-")
```